MailCore/Models/Recipient.swift

```diff
@@ -85,6 +91,25 @@ public final class Recipient: EmbeddedObject, Codable {
         return email == recipient.email && name == recipient.name
     }
 
+    public func isExternal(mailboxManager: MailboxManager) -> Bool {
+        ///if the email adress is added manually by me, it's not considered as an external
+        guard !isAddedByMe else { return false }
+
+        let trustedDomains = ["@infomaniak.com", "@infomaniak.event", "@swisstransfer.com"]
+        let isKnownDomain = trustedDomains.contains { domain in
+            return email.hasSuffix(domain)
+        }
+
+        guard let regex = Regex(pattern: "mailer-daemon@(?:.+.)?infomaniak.ch") else { return false }
```

Comment:
Tiny thing, I think we can put the pattern in a `static let`. 

Also I'm on the fence about adding a fatalError instead of a return in the else { } case